### ISSUE TITLE
Error with full path to secret

### ISSUE THREAD

**Issue by sd2020bs** [hashicorp/secrets-store-csi-driver-provider-vault#39]
When i create secrets in path /secret (fullpath to secrets is: /secret/database1.conf) and mount it  - all is ok.
But when i create secrets in path /secret/database  (fullpath to secrets is: /secret/database/test-db.conf ) i get error:
`Warning  FailedMount  3m55s  kubelet, k8-2      MountVolume.SetUp failed for volume "secrets-store" : rpc error: code = Unknown desc = error mounting secret time="2020-05-14T09:43:54Z" level=fatal msg="[error] : secrets-store csi driver failed to write /database/test-db.conf at /var/lib/kubelet/pods/14a6336a-4dc0-4d28-8cc1-23ffbca3aac0/volumes/kubernetes.io~csi/secrets-store/mount: open /var/lib/kubelet/pods/14a6336a-4dc0-4d28-8cc1-23ffbca3aac0/volumes/kubernetes.io~csi/secrets-store/mount/database/test-db.conf: no such file or directory"`

```
in vault policy i have been added:
path "secret/data/database/test-db.conf" {
  capabilities = ["read", "list"]
}
```

How can i use secrets with long name of dirs?

**Comment by F-Potter** [comment]
Have the same issue, seems like this is since the objectPath tries to mount it as /database/test-db.conf instead of database1.conf in /mnt/secrets-store. However, since /database doesn't exists this will not work.

I have tried to use the csi-secrets-store-provider-vault:0.0.5 in order to use the objectName as key, but this give the Mach-O file issue as mentioned earlier by others.

**Comment by mitsutaka** [comment]
I believe it is already solved by 0.0.5. binary format issue #38 has resolved.

**Comment by tomhjp** [comment]
As mentioned above, I think the solution here is to use `objectName` from version 0.0.5 onwards. 0.0.6 onwards will actually also support subpaths in `objectName`, creating folders that don't exist. Please feel free to open another issue about Mach-O issues in another issue if that's still happening - I'm not sure what that is.
